Cannot pull closed port (One2OneBidi.in)

When running HTTP without fusing

```
[info] [ERROR] [07/12/2016 17:49:53.307] [HttpBenchmark-akka.actor.default-dispatcher-25] [akka://HttpBenchmark/user/StreamSupervisor-1/flow-112-1-One2OneBidi] Error in stage [One2OneBidi]: requirement failed: Cannot pull closed port (One2OneBidi.in)
[info] java.lang.IllegalArgumentException: requirement failed: Cannot pull closed port (One2OneBidi.in)
[info] 	at scala.Predef$.require(Predef.scala:224)
[info] 	at akka.stream.stage.GraphStageLogic.pull(GraphStage.scala:347)
[info] 	at akka.http.impl.util.One2OneBidiFlow$One2OneBidi$$anon$1$$anon$4.onPush(One2OneBidiFlow.scala:84)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.processElement$1(GraphInterpreter.scala:590)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:601)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:542)
[info] 	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:471)
[info] 	at akka.stream.impl.fusing.GraphInterpreterShell.receive(ActorGraphInterpreter.scala:414)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:603)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:618)
[info] 	at akka.actor.Actor$class.aroundReceive(Actor.scala:484)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:529)
[info] 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
[info] 	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
[info] 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
[info] 	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
[info] 	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
[info] 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
[info] 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
[info] 	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
[info] 	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
[info]
[info] [ERROR] [07/12/2016 17:49:53.308] [HttpBenchmark-akka.actor.default-dispatcher-25] [akka.actor.ActorSystemImpl(HttpBenchmark)] Internal server error, sending 500 response
[info] java.lang.IllegalArgumentException: requirement failed: Cannot pull closed port (One2OneBidi.in)
[info] 	at scala.Predef$.require(Predef.scala:224)
[info] 	at akka.stream.stage.GraphStageLogic.pull(GraphStage.scala:347)
[info] 	at akka.http.impl.util.One2OneBidiFlow$One2OneBidi$$anon$1$$anon$4.onPush(One2OneBidiFlow.scala:84)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.processElement$1(GraphInterpreter.scala:590)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:601)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:542)
[info] 	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:471)
[info] 	at akka.stream.impl.fusing.GraphInterpreterShell.receive(ActorGraphInterpreter.scala:414)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:603)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:618)
[info] 	at akka.actor.Actor$class.aroundReceive(Actor.scala:484)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:529)
[info] 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
[info] 	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
[info] 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
[info] 	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
[info] 	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
[info] 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
[info] 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
[info] 	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
[info] 	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

--- a/akka-http-core/src/main/scala/akka/http/impl/util/One2OneBidiFlow.scala
+++ b/akka-http-core/src/main/scala/akka/http/impl/util/One2OneBidiFlow.scala
@@ -81,7 +81,7 @@ private[http] object One2OneBidiFlow {
             push(out, element)
             if (pullSuppressed) {
               pullSuppressed = false
-              pull(in)
+              if (!isClosed(in)) pull(in)
             }
           } else throw new UnexpectedOutputException(element)
         }

--- a/akka-http-core/src/test/scala/akka/http/impl/util/One2OneBidiFlowSpec.scala
+++ b/akka-http-core/src/test/scala/akka/http/impl/util/One2OneBidiFlowSpec.scala
@@ -113,6 +113,34 @@ class One2OneBidiFlowSpec extends AkkaSpec {
 
       out.sendComplete() // To please assertAllStagesStopped
     }
+
+    "not pull when input is closed before surpressed pull can be acted on" in assertAllStagesStopped {
+      val in = TestPublisher.probe[Int]()
+      val out = TestSubscriber.probe[Int]()
+      val wrappedIn = TestSubscriber.probe[Int]()
+      val wrappedOut = TestPublisher.probe[Int]()
+
+      Source.fromPublisher(in).via(
+        One2OneBidiFlow(maxPending = 1) join Flow.fromSinkAndSource(
+          Sink.fromSubscriber(wrappedIn),
+          Source.fromPublisher(wrappedOut))
+      ).runWith(Sink.fromSubscriber(out))
+
+      out.request(2)
+      wrappedOut.expectRequest()
+      wrappedIn.request(2)
+      in.expectRequest()
+      in.sendNext(1)
+      wrappedIn.expectNext(1)
+      // now we have reached the maxPending limit
+      in.sendComplete()
+      wrappedOut.sendNext(1)
+      out.expectNext(1)
+      wrappedIn.expectComplete()
+      wrappedOut.sendComplete()
+      out.expectComplete()
+
+    }
   }
 
   class Test(maxPending: Int = -1) {

